google/resource_storage_bucket.go

```diff
@@ -378,3 +448,135 @@ func flattenCors(corsRules []*storage.BucketCors) []map[string]interface{} {
 	}
 	return corsRulesSchema
 }
+
+func resourceGCSBucketLifecycleCreateOrUpdate(d *schema.ResourceData, sb *storage.Bucket, update bool) error {
+	if v, ok := d.GetOk("lifecycle_rule"); ok {
+		lifecycle_rules := v.([]interface{})
+
+		if len(lifecycle_rules) > 100 {
+			return fmt.Errorf("At most 100 lifecycle_rule blocks are allowed")
+		}
+
+		sb.Lifecycle = &storage.BucketLifecycle{}
+		sb.Lifecycle.Rule = make([]*storage.BucketLifecycleRule, 0, len(lifecycle_rules))
+
+		for _, lifecycle_rule := range lifecycle_rules {
+			lifecycle_rule := lifecycle_rule.(map[string]interface{})
+
+			target_lifecycle_rule := &storage.BucketLifecycleRule{}
+
+			if v, ok := lifecycle_rule["action"]; ok {
+				action := v.(*schema.Set).List()[0].(map[string]interface{})
+
+				target_lifecycle_rule.Action = &storage.BucketLifecycleRuleAction{}
+
+				if v, ok := action["type"]; ok {
+					target_lifecycle_rule.Action.Type = v.(string)
+				}
+
+				if v, ok := action["storage_class"]; ok {
+					target_lifecycle_rule.Action.StorageClass = v.(string)
+				}
+			}
+
+			if v, ok := lifecycle_rule["condition"]; ok {
+				condition := v.(*schema.Set).List()[0].(map[string]interface{})
+				condition_elements := 0
```

Comment:
Rather than counting the condition elements (which might be harder to track later) how about just asserting that the condition map contains at least one expected key?

Actually, correct me if I'm wrong, but can you just say something like
```
if len(condition) < 1 {
  // Error here
}
```